### PR TITLE
Improve coordinates (#32 and #5)

### PR DESCRIPTION
**Pull request by fenjalien**
I so far have parity with the current implementation, I want to have as many coordinate types from: https://tikz.dev/tikz-coordinates
As a list:
- [x] `xyz`
- [x] `polar`
- [x] `barycentric`
- [x] `node`
- [x] `perpendicular`
- [ ] `intersections`
- [x] `relative`
- [ ] `rotational relative`
- [ ] `modifiers`
    - [x] `partway`
    - [ ] `distance`
    - [ ] `projection`

Changes made so far:
- Explicit and implicit formats
- Relative coordinates with a don't update option (#5)
- Function format has to receive coordinates, other parameters should use `.with` (had to change image test)
- Refactored explicit anchor/node format

This shouldn't be merged until the manual has been updated to reflect these changes

### CODE REVIEW

**Comment by fenjalien** [comment]
Update: I think I have all the coordinate systems working.

Things I have yet to do: 
- [ ] Update the manual
- [ ] General tidy up
- [ ] Fix up error handling (#42)

I should note that the intersections coordinate system works slightly differently. When an element is named, its path drawable's coordinates are recorded in `ctx.nodes` (which is refactored from `ctx.anchors`). To get the intersection of two elements that contain paths, call the `intersections` draw function and pass it two named elements, anchors will then be created for every intersection of those two elements (`"intersections.1"` for example). In this form you can't iterate over an arbitrary number of intersections like you can with tikz. Getting that to work will take a bit longer.

**Comment by fenjalien** [comment]
CHANGE NOTES:
- Updated coordinate resolution system
- Added coordinate tyes
  - tangent
  - perpendicular
  - interpolation
  - barycentric
- Updated coordinate types
  - xyz
    - Renamed from absolute
    - Can specify individual directions e.g. `(x: 1)`
  - polar
    - Can use ellipse radii
  - relative
    - Use `update` to set whether or not the new position is updated
    - Use `to` to specify which position to be relative to instead of the previous
  - anchor
    - Renamed from node
- Added new draw element `grid`
- Fixed default anchor behaviours
- Added a function for all draw elements that accept coordinates to check if the coordinate is correct 
- Fixed demos
- Updated manual
  - Updated argument specification style
  - Updated example style
  - Updated elements heading styles
- Attempted to add intersection calculations but the recent update of using bezier curves for circles made things a bit difficult and out of scope for this PR.

@johannes-wolf, @PgBiel and @MultisampledNight how does this look?

**Comment by PgBiel** [comment]
Hey, thanks for working on this! I was trying to find time to open a PR for my proposal haha... I'll take a look and update when I can.

**Review by johannes-wolf — APPROVED** [review]
Looks awesome! I do not get how partway coordinates are handled, does it work for curves or is it linear?

File: coordinate.typ [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+        }
+      ),
+    1 / c.bary.values().sum()
+    )
+}
+
+#let resolve-relative(resolve, ctx, c) = {
+  // (rel: <coordinate>, update: <bool> or <none>, to: <coordinate>)
+  let update = c.at("update", default: true)
+  c = vector.add(resolve(ctx, c.rel), c.at("to", default: ctx.prev.pt))
```
[the remark this answers is not in the document]
Comment (fenjalien) [reply coordinate.typ]: woops good catch!

**Comment by fenjalien** [comment]
> Looks awesome! I do not get how partway coordinates are handled, does it work for curves or is it linear?

I've said "linearly interpolate" in the explanation as its supposed to be between two positions.

**Comment by johannes-wolf** [comment]
> > Looks awesome! I do not get how partway coordinates are handled, does it work for curves or is it linear?
> 
> I've said "linearly interpolate" in the explanation as its supposed to be between two positions.
> 
> @johannes-wolf

I think this is good enough for now. Supporting more complex paths is not that easy, because of the needed length calculation.

Is this ready to get merged?

**Comment by fenjalien** [comment]
YUP :))

**Pull request merged**